h2o-py/h2o/explanation/_explain.py

```diff
@@ -2225,16 +2309,31 @@ def _process_models_input(
     :returns: tuple (is_aml, models_to_show, classification, multinomial_classification,
                     multiple_models, targets, tree_models_to_show)
     """
-    is_aml = isinstance(models, h2o.automl._base.H2OAutoMLBaseMixin)
-    if is_aml:
-        models_to_show = [models.leader]
-        multiple_models = models.leaderboard.nrow > 1
+    is_aml = False
+    if _is_automl_or_leaderboard(models):
+        is_aml = True
+        if isinstance(models, h2o.automl._base.H2OAutoMLBaseMixin):
+            models_to_show = [models.leader]
+            models = models.leaderboard
+        else:
+            models_to_show = [h2o.get_model(models[0, "model_id"])]
+        if _has_varimp(models_to_show[0]):
+            models_with_varimp = models_to_show
+        else:
+            model_with_varimp = next(_get_models_from_automl_or_leaderboard(models, filter_=_has_varimp), None)
+            if model_with_varimp is None:
+                models_with_varimp = []
+            else:
+                models_with_varimp = [model_with_varimp]
```

Comment:
nitpick: one assignment better than 2
```suggestion
            models_with_varimp = [] if model_with_varimp is None else [model_with_varimp]
```